### PR TITLE
Create link-check.yml

### PR DESCRIPTION
**Pull request by korbinib**
Adds https://github.com/lycheeverse/lychee-action support for RDMkit
Will populate https://github.com/elixir-europe/rdmkit/issues/1570

### CODE REVIEW

File: .github/workflows/link-check.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    inputs:
+      logLevel:
+        description: 'Log level'
+        required: true
+        default: 'warning'
+        type: choice
+        options:
+        - info
+        - warning
+        - debug
```
Comment (bedroesb) [inline .github/workflows/link-check.yml:15]: Are these inputs needed? I think we can also remove repository dispatch

File: .github/workflows/link-check.yml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+      - name: Install dependencies
+        run: |
+            bundle install
+      - name: Build with Jekyll
+          # Outputs to the './_site' directory by default
+        run: |
+            bundle exec jekyll build
+        env:
+            PAGES_REPO_NWO: ${{ github.repository }}
+            JEKYLL_BUILD_BRANCH: ${{ github.ref_name }}
```
Comment (bedroesb) [inline .github/workflows/link-check.yml:46]:
```suggestion
            JEKYLL_BUILD_BRANCH: ${{ github.ref_name }}
            JEKYLL_ENV: production
```

File: .github/workflows/link-check.yml (hunk 1 above)
Comment (korbinib) [reply .github/workflows/link-check.yml:15]: no, will remove it.

**Review by bedroesb — APPROVED** [review]
Let's see what this gives!

**Pull request merged**